src/Microsoft.AspNetCore.Routing/Patterns/RoutePattern.cs

```diff
@@ -9,6 +9,11 @@
 
 namespace Microsoft.AspNetCore.Routing.Patterns
 {
+    /// <summary>
+    /// Represents a parts route template with default values and constraints.
+    /// Use <see cref="RoutePatternFactory"/> to create <see cref="RoutePattern"/>
+    /// instances.
+    /// </summary>
```

Comment:
Should it mention that it is immutable? Applies to all pattern types.

Reply:
Sure